### PR TITLE
i18n(fr): Adding `Deploy your Astro Site to GitHub Pages` translation

### PR DESCRIPTION
**Pull request by pioupia**
#### What kind of changes does this PR include?
- Translated content

#### Description
- Translated **Deploy your Astro Site to GitHub Pages** in French language. Updates content from `src/pages/fr/guides/deploy/github.md`

### CODE REVIEW

File: src/pages/fr/guides/deploy/github.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    on:
+      # Trigger the workflow every time you push to the `main` branch
+      # Using a different branch name? Replace `main` with your branch’s name
+      push:
+        branches: [ main ]
+      # Allows you to run this workflow manually from the Actions tab on GitHub.
+      workflow_dispatch:
+      
+    # Allow this job to clone the repo and create a page deployment
```
Comment (Yan-Thomas) [inline src/pages/fr/guides/deploy/github.md:37]: What do you think about translating these comments so the reader knows what it's happening without needing to know English? (you don't need to translate the action's steps)

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for this @pioupia, I added a small comment highlighting something that may be worth being translated.

File: src/pages/fr/guides/deploy/github.md (hunk 1 above)
Comment (pioupia) [reply src/pages/fr/guides/deploy/github.md:37]: Yeah, good idea ! I'll do that !

Comment (pioupia) [reply src/pages/fr/guides/deploy/github.md]: Done. https://github.com/withastro/docs/pull/1349/commits/64de601e30f17ea53b7017eecb76ded21a59941c

**Comment by Yan-Thomas** [comment]
Thanks, things are looking good to me! Just a thing, this page had some recent updates, could you add them to this PR so we can confidently merge it sure it's 100% up-to-date? https://github.com/withastro/docs/commit/7e4ffeedcb526e21d36a5444346323c49793d4f4

**Comment by pioupia** [comment]
Sure ! @Yan-Thomas

**Comment by pioupia** [comment]
Done !

**Comment by Yan-Thomas** [comment]
Thanks @pioupia, good job! I recommend translating the Deploy page next, people may think there's not deploy guides translated if the main deploy page isn't.

File: src/pages/fr/guides/deploy/github.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+
+    jobs:
+      build:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout your dépôt using git
```
[the remark this answers is not in the document]
Comment (pioupia) [reply src/pages/fr/guides/deploy/github.md:67]: No, it's a mistake... I'll correct that

File: src/pages/fr/guides/deploy/github.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          name: github-pages
+          url: ${{ steps.deployment.outputs.page_url }}
+        steps:
+          - name: Deploy to GitHub Pages
+            id: deployment
+            uses: actions/deploy-pages@v1
+    ```
+    
+    :::caution
+   [L'action](https://github.com/withastro/action) officielle Astro recherche un fichier de verrouillage pour détecter votre gestionnaire de paquets préféré (`npm`, `yarn`, ou `pnpm`). Vous devez commit le fichier généré automatiquement par votre gestionnaire de packages `package-lock.json`, `yarn.lock`, ou `pnpm-lock.yaml` dans votre dépôt.
```
[the remark this answers is not in the document]
Comment (pioupia) [reply src/pages/fr/guides/deploy/github.md:85]: and here it is ! [https://github.com/withastro/docs/pull/1532/](https://github.com/withastro/docs/pull/1532/)

[the remark this answers is not in the document]
Comment (thomasbnt) [reply src/pages/fr/guides/deploy/github.md:85]: Good!

**Pull request merged**